injected/src/features/duck-ai-data-clearing.js

```diff
@@ -0,0 +1,96 @@
+import ContentFeature from '../content-feature.js';
+
+/**
+ * This feature is responsible for clearing Duck.ai-related data when the `duckAiClearData`
+ * message is received. It clears the `savedAIChats` item from localStorage and the `chat-images`
+ * object store from IndexedDB, then sends a `duckAiClearDataCompleted` message if successful
+ * or a `duckAiClearDataFailed` message if either call is unsuccessful.
+ */
+export class DuckAiDataClearing extends ContentFeature {
+    init() {
+        this.messaging.subscribe('duckAiClearData', (_) => this.clearData());
+    }
+
+    async clearData() {
+        let success = true;
+
+        try {
+            this.clearSavedAIChats();
+        } catch (error) {
+            success = false;
+            this.log.error('Error clearing saved chats:', error);
+        }
+
+        try {
+            await this.clearChatImagesStore();
+        } catch (error) {
+            success = false;
+            this.log.error('Error clearing saved chat images:', error);
+        }
+
+        if (success) {
+            this.notify('duckAiClearDataCompleted');
+        } else {
+            this.notify('duckAiClearDataFailed');
+        }
+    }
+
+    clearSavedAIChats() {
+        const localStorageKey = this.getFeatureSetting('chatsLocalStorageKey');
```

Comment:
Is it worth making this a list now just in case there ends up being adjacent data later? Only needing a config update would be nice.